### PR TITLE
mzbuild: fix double compile bug

### PR DESCRIPTION
**Pull request by benesch**
CargoBuild.build invokes `cargo build` in *two* places. Be sure to pass
the same compilation flags in both places to avoid thrashing the cache.
This should substantially speed up builds in CI and local development.

### CODE REVIEW

File: misc/python/materialize/cli/mzimage.py [hunk 1]
```diff
@@ -84,6 +88,17 @@ def add_image_subcommand(name: str, **kwargs: Any) -> argparse.ArgumentParser:
...
+        subparser.add_argument(
+            "--build-mode",
+            default="release",
+            choices=["dev", "release"],
+            help="whether to build in dev or release mode",
+        )
+        subparser.add_argument(
+            "--coverage",
+            help="whether to enable code coverage compilation flags",
+            action="store_true",
```
Comment (aljoscha) [inline misc/python/materialize/cli/mzimage.py:100]: In `cli/mzcompose.py` we also add `default=None` here, but I imagine that's the, ahem, default anyways?

File: misc/python/materialize/cli/mzimage.py [hunk 2]
```diff
@@ -84,6 +88,17 @@ def add_image_subcommand(name: str, **kwargs: Any) -> argparse.ArgumentParser:
         subparser.add_argument(
             "image", help="the name of an mzbuild image in this repository"
         )
+        subparser.add_argument(
```
Comment (aljoscha) [inline misc/python/materialize/cli/mzimage.py:91]: `mzcompose` calls these arguments `--mz-build-mode` and `--mz-coverage`, respectively. I'm not sure about the naming scheme of these tools but using the same names here would make it less confusing for me at least. 😅

**Review by aljoscha — COMMENTED** [review]
The invocation now matches the other cargo_build invocation, so it should work. Though I haven't tested yet. I can do that once I'm on my bigger machine.

I had a comment about argument naming.

Also, thanks for fixing this! 😃 (it's weird that we were accepting it as "normal" for so long)

File: misc/python/materialize/cli/mzimage.py (hunk 2 above)
Comment (benesch) [reply misc/python/materialize/cli/mzimage.py:91]: Ah yeah, agree it's confusing! mzcompose does that to distinguish between arguments passed to Docker Compose and arguments passed to mzcompose. It's probably a mistake, and we should strip the "mz-" prefix from those commands. We're slowly moving towards hiding Docker Compose away.

File: misc/python/materialize/cli/mzimage.py (hunk 1 above)
Comment (benesch) [reply misc/python/materialize/cli/mzimage.py:100]: Ah yeah, I wonder why I typed that `default=None` explicitly in mzcompose...

**Comment by benesch** [comment]
Thanks very much! I think it's only been broken like this for a couple days, actually.

**Pull request merged**